lib/ASTGen/Sources/ASTGen/ASTGen.swift

```diff
@@ -138,6 +138,73 @@ struct ASTGenVisitor {
   }
 }
 
+extension ASTGenVisitor {
+  /// Obtains a bridged, `ASTContext`-owned "identifier".
+  ///
+  /// If the token text is `_`, return an empty identifier. If the token is an
+  /// escaped identifier, backticks are stripped.
+  func bridgedIdentifier(token: TokenSyntax) -> BridgedIdentifier {
+    var text = token.rawText
+    if text == "_" {
+      return nil
+    }
+    if text.count > 2 && text.hasPrefix("`") && text.hasSuffix("`") {
+      text = .init(rebasing: text.dropFirst().dropLast())
+    }
+    return self.ctx.getIdentifier(text.bridged)
+  }
+
+  /// Obtains a bridged, `ASTContext`-owned "identifier".
+  ///
+  /// If the `token` text is `nil`, return an empty identifier.
+  func bridgedIdentifier(token: TokenSyntax?) -> BridgedIdentifier {
+    token.map(bridgedIdentifier(token:)) ?? nil
+  }
+
+  /// Obtains the start location of the node excluding leading trivia in the
+  /// source buffer.
+  func bridgedSourceLoc(syntax node: some SyntaxProtocol) -> BridgedSourceLoc {
+    BridgedSourceLoc(at: node.positionAfterSkippingLeadingTrivia, in: self.base)
+  }
```

Comment:
IMO the `token:` and `syntax:` argument labels aren't adding much useful information, could they be replaced by `for:`?

Reply:
Also the old variants of all these methods were marked `@inline(__always)`, does that need to be preserved?